### PR TITLE
 GH-4189: Added LS start-up error handler.

### LINKED ISSUE #2914: python file has no highlight or prompt
I wan to use Theia to edit a python file. However, I found it has no highlight nor prompt of input. Could anyone please help on this?

image

### LINKED ISSUE #4189: [java] The Java backend should warn the user when Java is not on the `PATH`
It fails silently now.

### PR DESCRIPTION
**Pull request by kittaakos**
Closes #4189.
Closes #2914.

Signed-off-by: Akos Kitta <kittaakos@typefox.io>

### CODE REVIEW

File: packages/languages/src/node/language-server-contribution.ts [hunk 1]
```diff
@@ -76,23 +75,54 @@ export abstract class BaseLanguageServerContribution implements LanguageServerCo
...
     }
 
+    protected spawnProcessAsync(command: string, args?: string[], options?: cp.SpawnOptions): Promise<RawProcess> {
+        const rawProcess = this.processFactory({ command, args, options });
+        rawProcess.process.stderr.on('data', this.logError.bind(this));
+        return new Promise<RawProcess>((resolve, reject) => {
+            rawProcess.onError(error => {
+                this.onDidFailSpawnProcess(error);
+                if (isErrnoException(error) && error.code === 'ENOENT') {
+                    const guess = command.split('\S').shift();
```
Comment (simark) [inline packages/languages/src/node/language-server-contribution.ts]: What is this line meant to do?  It looks like you wanted to do a regex, but it isn't a RegExp object, so I think it doesn't do anything for now.

Comment (kittaakos) [reply packages/languages/src/node/language-server-contribution.ts]: extract the command.

Comment (simark) [reply packages/languages/src/node/language-server-contribution.ts]: And does it work?

Comment (kittaakos) [reply packages/languages/src/node/language-server-contribution.ts]: I guess no, it should be `split(/\s/).shift()`.

**Pull request merged**